### PR TITLE
event grid namespaces infrastructure and access

### PR DESCRIPTION
**Pull request by geoberle**
this PR introduces a bicep templates to create the regional cloud infrastructure required by maestro and defined in [SD-DDR-0024](https://docs.google.com/document/d/1JUbv0Zco--SPpWH7pIsuzuKTUtq2S4WovAoPuAt5PPI/edit).

the general infrastructure part is managed in the `maestro-infra.bicep` module. it is included into the `svc-cluster.bicep` template as proposed in SD-DDR-0030.

* create an eventgrid MQTT broker
* creates a keyvault for the client certificates that are used for authentication
  * right now, only self-signed certificates are supported, which is fine for DEV
  * for int/prod OneCertV2 will be used
* defines the basic authentication configuration for eventgrid (MQTT client groups, topic spaces, topic access templates)

authentication and authorization for a maestro server or consumer is managed via the `maestro-eventgrid-access.bicep` module. it is responsible for:

* creating a client certificate for authn in key vault
* grants key vault access to the certificate to a managed identity (used for CSI secret store to access the secret from the cluster)
* registers an MQTT client in the EventGrid client registry and assigns proper access

the `maestro-server.bicep` module dresses up a Service Cluster for the installation of the Maestro server. it leverages `maestro-eventgrid-access.bicep` to define the broker access and places some manifests on the AKS cluster:

* a `Secret` containing EventGrid details like the hostname
* a CSI `SecretProviderClass` CR to access the client certificate for broker access

part of https://issues.redhat.com/browse/ARO-7244

### What this PR does
introduce bicep config for maestro regional infrastructure.

Jira: https://issues.redhat.com/browse/ARO-7244

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Deployment: The deployment process was considered and addressed if required
- [ ] Testing: New code requires new unit tests.
- [ ] Documentation: Is the documentation updated? Either in the doc located in focus area, in the README or in the code itself.
- [ ] Customers: Is this change affecting customers? Is the release plan considered?

### CODE REVIEW

**Comment by geoberle** [comment]
@microsoft-github-policy-service agree company="Red Hat"

File: dev-infrastructure/templates/svc-cluster.bicep [hunk 1]
```diff
@@ -29,8 +29,13 @@ param disableLocalAuth bool
 @description('Deploy ARO HCP RP Azure Cosmos DB if true')
 param deployFrontendCosmos bool
 
+<<<<<<< HEAD
```
Comment (tonytheleg) [inline dev-infrastructure/templates/svc-cluster.bicep:32]: Looks like some rebase artifacts that just need some adjusting here

File: dev-infrastructure/templates/svc-cluster.bicep [hunk 2]
```diff
@@ -65,3 +68,29 @@ module rpCosmosDb '../modules/rp-cosmos.bicep' =
...
+}
+
+module maestroServer '../modules/maestro-eventgrid-access.bicep' = if (deployMaestroInfra) {
+  name: 'maestro-server-eventgrid-access'
+  scope: resourceGroup()
+  params: {
+    eventGridNamespaceName: maestroInfra.outputs.eventGridNamespaceName
+    keyVaultName: maestroInfra.outputs.keyVaultName
+    keyVaultManagedIdentityId: maestroInfra.outputs.keyVaultCertificateOfficerManagedIdentityId
+    certDomain: 'maestro.${location}.azure.com'
```
Comment (geoberle) [inline dev-infrastructure/templates/svc-cluster.bicep:88]: consider this a placeholder. the details about the oncert domains to use will be determined in the DDR

File: dev-infrastructure/templates/svc-cluster.bicep (hunk 1 above)
Comment (geoberle) [reply dev-infrastructure/templates/svc-cluster.bicep]: fixed. thank you!

File: dev-infrastructure/modules/aks-cluster-base.bicep [hunk 3]
```diff
@@ -253,7 +253,7 @@ resource aksCluster 'Microsoft.ContainerService/managedClusters@2024-01-01' = {
         enabled: true
         config: {
           enableSecretRotation: 'true'
-          rotationPollInterval: '24h'
+          rotationPollInterval: '5m'
```
Comment (mjlshen) [inline dev-infrastructure/modules/aks-cluster-base.bicep:256]: Is this an intentional change? The change makes sense to me, I don't know why it was 24h previously, but just want to make sure

**Review by mjlshen — APPROVED** [review]

**Review by mjlshen — APPROVED** [review]

File: dev-infrastructure/modules/aks-cluster-base.bicep (hunk 3 above)
Comment (geoberle) [reply dev-infrastructure/modules/aks-cluster-base.bicep:256]: we can discuss around the value. i think the default is 2m

**Review by tonytheleg — APPROVED** [review]

**Pull request merged**